opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java

```diff
@@ -1347,17 +1407,24 @@ public void writeConfiguration(File file) throws IOException {
      * @throws IOException if an error occurs
      */
     public void writeConfiguration(String host) throws IOException {
-        Response r = ClientBuilder.newClient()
-                .target(host)
-                .path("api")
-                .path("v1")
-                .path("configuration")
-                .queryParam("reindex", true)
-                .request()
-                .put(Entity.xml(configuration.getXMLRepresentationAsString()));
-
-        if (r.getStatusInfo().getFamily() != Response.Status.Family.SUCCESSFUL) {
-            throw new IOException(r.toString());
+        Lock readLock = configLock.readLock();
+        try {
+            readLock.lock();
+
+            Response r = ClientBuilder.newClient()
+                    .target(host)
+                    .path("api")
+                    .path("v1")
+                    .path("configuration")
+                    .queryParam("reindex", true)
+                    .request()
+                    .put(Entity.xml(configuration.getXMLRepresentationAsString()));
```

Comment:
Is it neccessary to lock the whole api call which can take a non-trivial amount of time while the only thing which needs the lock is this `configuration.getXMLRepresentationAsString()`